common/src/test/org/apache/hadoop/hive/conf/TestHiveConf.java

```diff
@@ -252,4 +254,16 @@ public void testAdditionalConfigFiles() throws Exception{
     f2.delete();
     fileBakHiveSite.delete();
   }
+
+  @Test
+  public void testGetMatchingEntries() {
+    HiveConf conf = new HiveConf();
+    conf.setInt("hive.compactor.worker.iceberg.threads", 4);
+
+    List<Map.Entry<String, String>> entries = conf.getMatchingEntries(Constants.COMPACTION_POOLS_PATTERN);
```

Comment:
why is that a `List<Map.Entry<String, String>>` and not just Map<String, String>?

Reply:
that is some crazy code, could you please refactor it with
````
  public Map<String, Integer> getPoolConfigEntries(HiveConf hiveConf) {
    Map<String, Integer> poolConfigs = new HashMap<>();
    for (Map.Entry<String, String> entry : hiveConf) {
      Matcher matcher = COMPACTION_POOLS_PATTERN.matcher(entry.getKey());
      if (matcher.matches()) {
        matchingEntries.put(matcher.group(0), NumberUtils.toInt(entry.getValue(), 0));
      }
    }
    return poolConfigs;
  }
````